src/java/org/apache/cassandra/config/Config.java

```diff
@@ -430,6 +430,12 @@ public static class SSTableConfig
     public DurationSpec.IntMillisecondsBound dynamic_snitch_reset_interval = new DurationSpec.IntMillisecondsBound("10m");
     public double dynamic_snitch_badness_threshold = 1.0;
 
+    public String node_proximity;
+    public String initial_location_provider;
+    public String addresses_config;
+    // TODO should this be true by default? may be a change in behaviour for anything other than EC2MRSnitch
```

Comment:
yep maybe

Same with the new default method I added in IEndpointSnitch `preferLocalConnections` - maybe we should make that non-default to force any custom snitches to update?

Reply:
I think the issue with doing that it makes the deprecation cycle a bit weird because we're marking `IEndpointSnitch` as deprecated while at the same time requiring maintainers to update their implementations.
Making them non-default is stricter/more safe and I don't know how effective a NEWS.txt item alone would be.